### PR TITLE
docs(metadata-service): refine data model

### PR DESCRIPTION
**Pull request by BalduinLandolt**
resolves BS-41

Refine data model, create a json schema and UML diagram

### CODE REVIEW

**Comment by subotic** [comment]
@BalduinLandolt here are my answers to the questions in the datamodel.md and also some comments (in no particular order):

As a general remark, I like to model everything as loosely coupled as possible. There are only two types of values, either value objects or instances of classes. If it has more than one property or it has a chance to be extended in the future, then it is a class.

- Class: Grant: is missing `id` and `type`. I see grant as a separate thing.

**Comment by BalduinLandolt** [comment]
@subotic @mpro7 not much has changed since last review. But I think it's finalized now.

File: docs/services/metadata/example.json [hunk 1]
```diff
@@ -0,0 +1,271 @@
+{
```
Comment (mpro7) [inline docs/services/metadata/example.json:1]: Do we need this file anymore if there is `api-example.puml` which I think is 1:1 with this one?

**Review by mpro7 — CHANGES_REQUESTED** [review]

File: docs/services/metadata/datamodel.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+## API representation
+
+A metadata set can be serialized to json. This json data must conform to the [json-schema](schema-metadata.json).  
+The following example illustrates how the json might look:
+
+![json example](./api-example.svg)  
+(Click on the image and open `raw` for a reasonable zoom level.)  
+(Cf. `example.json` too.)
+
+The json representation is "flat", i.e. not nested, so all top-level types are present in the first level of depth of the json document tree. All those objects have a unique `@id` property. Wherever this object is referenced further down in the document, this is done so by this ID.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/services/metadata/datamodel.md:21]: I don't think so, but I could be wrong

File: docs/services/metadata/datamodel.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+The json representation is "flat", i.e. not nested, so all top-level types are present in the first level of depth of the json document tree. All those objects have a unique `@id` property. Wherever this object is referenced further down in the document, this is done so by this ID.  
+(NB: json schema does not allow for consistency checks of internal references, so the existence of an object with a given ID can not be guaranteed by json validation.)
+
+This json contains the entirety of a dataset. Endpoints can be implemented that return only part of the data, e.g. only the project, a list of all datasets, a dataset with a certain ID, etc. However, these endpoints should return an exact subset of the data i the full representation of the metadata.
+
+Other serializations of the data are possible. Eventually, a serialization in RDF should bee added.
+
+### Macro structure
+
+The overall structure of the json representation of a metadata-set should look like this:
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/services/metadata/datamodel.md:30]: you're right

File: docs/services/metadata/datamodel.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpro7) [reply docs/services/metadata/datamodel.md:21]: I'm sorry for my poor English 👀 , you're right.

[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/services/metadata/datamodel.md]: all good! :)

File: docs/services/metadata/api-example.puml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+        "url": "https://www.geonames.org/6255148/europe.html"
+      }
+    ],
+    "urls": [
+      {
+        "propertyID": "DaSCH.swiss",
+        "type": "URL",
+        "url": "https://dasch.swiss/someproject"
+      }
+    ],
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/services/metadata/api-example.puml:50]: done, renamed to `text`

**Review by mpro7 — APPROVED** [review]
Thanks!

**Pull request merged**